src/read.jl

```diff
@@ -186,12 +202,35 @@ end
 """
     EDF.read(file::AbstractString)
 
-Read the given file and return an `EDF.File` object containing the parsed data.
+Read the given file and return an `EDF.File` object containing all parsable
+header, signal, and annotation data in the EDF file at `path`.
 """
-function read(file::AbstractString)
-    open(file, "r") do io
-        file_header, signal_headers = read_file_and_signal_headers(io)
-        signals, annotations = read_signals(io, file_header, signal_headers)
-        return File(file_header, signals, annotations)
-    end
+read(path::AbstractString) = read!(open(path))
+
+"""
+    EDF.read!(file::File)
+
+Read the sample values and annotation data for `file.signals` and `file.annotations`
+from `file.io` and add them to `file`, returning the modified file. For files that
+have already been completely read, this function returns the identity.
+"""
+function read!(file::File)
+    (isopen(file.io) || !eof(file.io)) && read_signals!(file)
+    return file
+end
+
+"""
+    EDF.open(path::AbstractString)
+
+Read the file, signal, and annotation-wide file headers from the EDF file at
+`path`, returning an `File` instance without reading the file's sample data.
```

Comment:
```suggestion
`path`, returning a `File` instance without reading the file's sample data.
```

Reply:
ah, the old "used to be `EDF.File`" doc error. fixing!

Reply:
Fixed in https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/6163c31f3a4cb42399643f92820edb8950b670d7